### PR TITLE
Healthcheck: Aggregations

### PR DESCRIPTION
**Pull request by chao-xian**
## What

Raises `2ndline` alarms, following the similar approach to `daily_metrics_check.rb`, to notify when metric aggregations collected from `Etl::Aggregations::Monthly.process(date: date)` are missing.

## Why 
If the metrics are not present, then the ETL master process failed, and 2nd line needs to take an action

### CODE REVIEW

**Comment by chao-xian** [comment]
[(2) Raise 2ndline alarms when the Monthly aggregations are not collected from Google](https://trello.com/c/7Z5SLv8A/1141-2-raise-2ndline-alarms-when-the-monthly-aggregations-are-not-collected-from-google)

File: app/domain/healthchecks/aggregations.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    end
+
+    def message
+      "ETL :: no aggregations of metrics for yesterday" if status == :critical
+    end
+
+  private
+
+    def aggregations
+      @aggregations ||= Finders::Aggregations.new.between(from: Date.yesterday, to: Date.today).run
```
Comment (pmanrubia) [inline app/domain/healthchecks/aggregations.rb]: Not sure about the Aggregations. We would need to ensure that we have monthly aggregations. I would count on this table filtering by creation time.

Comment (chao-xian) [reply app/domain/healthchecks/aggregations.rb]: Addressed in https://github.com/alphagov/content-performance-manager/pull/1153/commits/69c1cacfa050fc37d3341465f047b3e0f3edec4c

File: app/domain/healthchecks/monthly_aggregations.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    end
+
+    def message
+      "ETL :: no aggregations of metrics for yesterday" if status == :critical
+    end
+
+  private
+
+    def aggregations
+      @aggregations ||= Aggregations::MonthlyMetric.where(created_at: Date.yesterday).count
```
Comment (pmanrubia) [inline app/domain/healthchecks/monthly_aggregations.rb]: Still, although the ETL process runs daily and retrieves the metrics for the day before, it seems to me that the `created_at` attribute should be `today`?

Could you please add an index for this field as well?

File: spec/domain/healthchecks/monthly_aggregations_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  include_examples 'Healthcheck enabled/disabled within time range'
+
+  around do |example|
+    Timecop.freeze(Date.new(2019, 2, 22)) { example.run }
+  end
+
+  its(:name) { is_expected.to eq(:aggregations) }
+
+  describe '#status' do
+    context 'When there are aggregations' do
```
Comment (pmanrubia) [inline spec/domain/healthchecks/monthly_aggregations_spec.rb]: Could you please make it explicit that they are `monthly` aggregations?

File: spec/domain/healthchecks/monthly_aggregations_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      before do
+        date = Date.yesterday
+        edition = create :edition, date: date
```
Comment (pmanrubia) [inline spec/domain/healthchecks/monthly_aggregations_spec.rb]: There is no need to set `date` for the `Edition`, we only need it for the metric.

Comment (chao-xian) [reply spec/domain/healthchecks/monthly_aggregations_spec.rb]: Thanks - addressed in https://github.com/alphagov/content-performance-manager/pull/1153/commits/6a00f3cdfb70e20e266eb93af9c07911eb28a6da

File: spec/domain/healthchecks/monthly_aggregations_spec.rb (hunk 3 above)
Comment (chao-xian) [reply spec/domain/healthchecks/monthly_aggregations_spec.rb]: Thanks - addressed in https://github.com/alphagov/content-performance-manager/pull/1153/commits/01a92df30f727fd33857bba12d4ada76cd6471ae

File: app/domain/healthchecks/monthly_aggregations.rb (hunk 2 above)
Comment (chao-xian) [reply app/domain/healthchecks/monthly_aggregations.rb]: Thanks - addressed in https://github.com/alphagov/content-performance-manager/pull/1153/commits/01a92df30f727fd33857bba12d4ada76cd6471ae and added index in https://github.com/alphagov/content-performance-manager/pull/1153/commits/5adefb394a85ae7c3a6fd9872cd16c4330695525

File: app/domain/healthchecks/monthly_aggregations.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    def name
+      :aggregations
+    end
+
+    def status
+      aggregations.positive? ? :ok : :critical
+    end
+
+    def message
+      "ETL :: no aggregations of metrics for yesterday" if status == :critical
```
Comment (pmanrubia) [inline app/domain/healthchecks/monthly_aggregations.rb]: Could you please change the message to `...no monthly aggregations...`

Comment (chao-xian) [reply app/domain/healthchecks/monthly_aggregations.rb]: Thanks @pmanrubia - [fixed](https://github.com/alphagov/content-performance-manager/pull/1153/commits/5f2a433ed28162d505c9dc798180201753d2351d)

**Comment by pmanrubia** [comment]
👍

**Pull request merged**